third_party/move/tools/move-asm/src/disassembler.rs

```diff
@@ -0,0 +1,705 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Disassembler for Move bytecode.
+
+use anyhow::bail;
+use move_binary_format::{
+    access::ModuleAccess,
+    file_format::{
+        Bytecode, CodeOffset, CompiledScript, FieldHandleIndex, LocalIndex, SignatureIndex,
+        SignatureToken, VariantFieldHandleIndex, VariantIndex, Visibility,
+    },
+    module_script_conversion::script_into_module,
+    views::{
+        FieldDefinitionView, FunctionDefinitionView, FunctionHandleView, ModuleView,
+        StructDefinitionView, StructHandleView, ViewInternals,
+    },
+    CompiledModule,
+};
+use move_core_types::{
+    ability::AbilitySet,
+    identifier::Identifier,
+    language_storage::{pseudo_script_module_id, ModuleId},
+};
+use std::{
+    cmp::Ordering,
+    collections::{BTreeMap, BTreeSet},
+    fmt,
+};
+
+pub fn disassemble_module<T: fmt::Write>(out: T, module: &CompiledModule) -> anyhow::Result<T> {
+    Disassembler::run(out, module)
+}
+
+pub fn disassemble_script<T: fmt::Write>(out: T, script: &CompiledScript) -> anyhow::Result<T> {
+    let script_as_module = script_into_module(script.clone(), "main");
```

Comment:
Does the assembler call the entrypoint as "main"? Do not see any tests for scrips and do not remember what name was used by assembler for compiled modules? If it is "main", maybe re-use as some constant?